### PR TITLE
CI Visibility GitLab Failure Reasons Table

### PR DESCRIPTION
**Pull request by kassenq**
### What does this PR do?
This PR adds a sub-section under the "Visualize pipeline data in Datadog" section on [this page](https://docs.datadoghq.com/continuous_integration/setup_pipelines/gitlab/?tab=gitlabcom). Under this sub-section, there is an image of failure reasons displayed within the product and a table that correlates error types to error messages and their source domain. 

### Motivation

We recently shipped a feature within Pipeline Visibility that helps users understand more about their GitLab pipeline errors and where they are coming from. Since this is only available for GitLab, the section was added on the page for tracing GitLab pipelines.
The table was added because GitLab has a plethora of error types, so we wanted to explicitly call out in the documentation which errors we support and what error message the user will be seeing in-app that corresponds to that error. We also provide information for whether that error is from the CI provider (GitLab) or the user.

### Preview
New or renamed files
https://docs-staging.datadoghq.com/kassen/error_table/integrations/iam_access_analyzer
Modified Files
https://docs-staging.datadoghq.com/kassen/error_table/continuous_integration/setup_pipelines/gitlab

### Additional Notes
None

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/setup_pipelines/gitlab.md [hunk 1]
```diff
@@ -149,6 +149,48 @@ through the UI by going to **Settings > CI/CD > Runners** and editing the approp
 After these steps, CI Visibility adds the hostname to each job. To see the metrics, click on a job span in the trace
 view. In the drawer, a new tab named **Infrastructure** appears which contains the host metrics.
 
+### Error messages for pipeline failures
+
+For failed GitLab pipeline executions (on the [Pipeline Executions][5] page), each error under the `Errors` tab within a specific execution displays a message associated with the error type from GitLab.
```
Comment (KickingTheTV) [inline content/en/continuous_integration/setup_pipelines/gitlab.md:154]: "For failed GitLab pipeline executions (on the Pipeline Executions page)" sounds like you can only see this from the Pipeline Executions page you've linked to.

Can you re-word to make it clear that users will always be able to see this whenever they go into any pipeline execution and that the page linked is one of which places you can see it?

Comment (kassenq) [reply content/en/continuous_integration/setup_pipelines/gitlab.md]: Removed the link as I think it made things more confusing. Thanks!

File: content/en/integrations/iam_access_analyzer.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
+---
```
Comment (urseberry) [inline content/en/integrations/iam_access_analyzer.md:1]: Is this file supposed to be included in this PR? Your comments only mention the new section on the "Set up Tracing on a GitLab Pipeline" page.

Comment (kassenq) [reply content/en/integrations/iam_access_analyzer.md:1]: Hi Ursula, this file is not supposed to be included in this PR - not sure how I managed to add an entire new page of content (might have pulled from somewhere else by accident)

**Review by urseberry — APPROVED** [review]
I will merge this when the build preview succeeds.

**Pull request merged**